server/subscribe.go

```diff
@@ -98,71 +113,68 @@ func (p *Plugin) getUserID() string {
 	return p.getConfig().botUserID
 }
 
-func (p *Plugin) getChannelsSubscribed(jwh *JiraWebhook) ([]string, error) {
+func (p *Plugin) getChannelsSubscribed(wh Webhook, jwh *JiraWebhook) ([]string, error) {
```

Comment:
I'm not certain, but don't we always attach the original jwh in the wh struct? If so we only need to pass the wh, right?

Reply:
Right, thanks for showing this :+1:

Reply:
On second look, both are being passed here because this is not explicitly the struct `webhook`, but the interface `Webhook` which does not expose a `JiraWebhook` struct. So we either expose it with a method, or make this function take the more private `webhook`.

I'm glad to make this change, I'm thinking we expose the `JiraWebhook` through a method `GetJiraWebhook()`. How does this sound @cpoile?

Reply:
Changed to use `GetJiraWebhook()`

Reply:
I'm 1/5 for not making a new method and just changing the param to `*webhook` -- I've been doing the same in the webhook parser, and I think it makes sense to keep passing the struct with the most information (rather than narrowing down what you pass, and then calling a function to expose what's already there).

Reply:
@cpoile Would you be okay with a type conversion like this to allow the function to access the JiraWebhook directly:
`channelIds, err :=  p.getChannelsSubscribed(wh.(*webhook))`?

Otherwise, the return value of all of the functions in `webhook_parser.go` need to change from `Webhook` to `*webhook`.

Reply:
I think if we need access to the webhook struct, just return it when we need it (that's what I've been doing while editing `webhook_parser`). But casting `Webhook` works too, since we know what's underneath. So it's your call.

Reply:
@cpoile Updated with type conversion